libkkc: FTBFS in Fedora rawhide/f31

Originally reported in Fedora 31 bugzilla; https://bugzilla.redhat.com/show_bug.cgi?id=1736028

Seems libkkc build is failed with the latest vala 0.45.3 .

The following is the build log:
```
BUILDSTDERR: metadata-file.vala:46.9-46.27: error: Creation method of abstract class cannot be public.
BUILDSTDERR:         public MetadataFile (string name, string filename) throws Error {
BUILDSTDERR:         ^^^^^^^^^^^^^^^^^^^
BUILDSTDERR: language-model.vala:104.9-104.28: error: Creation method of abstract class cannot be public.
BUILDSTDERR:         public LanguageModel (LanguageModelMetadata metadata) throws Error {
BUILDSTDERR:         ^^^^^^^^^^^^^^^^^^^^
BUILDSTDERR: make[2]: *** [Makefile:1429: libkkc_la_vala.stamp] Error 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -5,7 +5,7 @@ language: c
 env:
   - BUILD_OPTS="--prefix=/usr --libdir=/usr/lib64"
   - BUILD_OPTS="--prefix=/usr --libdir=/usr/lib64 --enable-docs" DOCS=yes EXTRA_PKGS="gtk-doc valadoc valadoc-devel"
-  - BUILD_OPTS="--prefix=/usr --libdir=/usr/lib64 --enable-code-coverage" COVERAGE=yes EXTRA_PKGS="python-pip"
+  - BUILD_OPTS="--prefix=/usr --libdir=/usr/lib64 --enable-code-coverage PYTHON=python3" COVERAGE=yes EXTRA_PKGS="python3-marisa python3-pip"
   - BUILD_OPTS="--prefix=/usr --libdir=/usr/lib64 CFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' CXXFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' LDFLAGS='-fsanitize=address -g -fno-common -U_FORTIFY_SOURCE' LIBS='-ldl -lpthread'" EXTRA_PKGS="libasan"
   - BUILD_OPTS="--prefix=/usr --libdir=/usr/lib64 CFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' CXXFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' LDFLAGS='-fsanitize=undefined -g -fno-common -U_FORTIFY_SOURCE' LIBS='-ldl -lpthread'" EXTRA_PKGS="libubsan"
 

--- a/configure.ac
+++ b/configure.ac
@@ -125,7 +125,7 @@ VALADOC_REQUIRED=0.3.1
 have_valadoc=no
 if test x$enable_docs = xyes; then
   # make sure the library is new enough and the program exists
-  PKG_CHECK_MODULES([VALADOC], [valadoc-0.40 >= $VALADOC_REQUIRED])
+  PKG_CHECK_MODULES([VALADOC], [valadoc-0.44 >= $VALADOC_REQUIRED])
   AC_PATH_PROG([VALADOC], [valadoc], [:])
   AS_IF([test "$VALADOC" != :], have_valadoc=yes)
 fi

--- a/libkkc/language-model.vala
+++ b/libkkc/language-model.vala
@@ -101,7 +101,7 @@ namespace Kkc {
         public abstract new LanguageModelEntry? @get (string input,
                                                       string output);
 
-        public LanguageModel (LanguageModelMetadata metadata) throws Error {
+        protected LanguageModel (LanguageModelMetadata metadata) throws Error {
             Object (metadata: metadata);
             init (null);
         }

--- a/libkkc/metadata-file.vala
+++ b/libkkc/metadata-file.vala
@@ -43,7 +43,7 @@ namespace Kkc {
          */
         public string filename { get; construct set; }
 
-        public MetadataFile (string name, string filename) throws Error {
+        protected MetadataFile (string name, string filename) throws Error {
             Object (name: name, filename: filename);
             init (null);
         }

--- a/tests/lib/test-case.vala
+++ b/tests/lib/test-case.vala
@@ -29,7 +29,7 @@ public abstract class Kkc.TestCase : Object
 
   public delegate void TestMethod ();
 
-  public TestCase (string name)
+  protected TestCase (string name)
     {
       this._suite = new GLib.TestSuite (name);
     }

